core/src/main/scala/org/apache/spark/sql/delta/DeltaTable.scala

```diff
@@ -356,4 +356,14 @@ object DeltaTableUtils extends PredicateHelper
   def parseColToTransform(col: String): IdentityTransform = {
     IdentityTransform(FieldReference(Seq(col)))
   }
+
+  def withActiveSession[T](spark: SparkSession)(body: => T): T = {
+    val old = SparkSession.getActiveSession
+    SparkSession.setActiveSession(spark)
+    try {
+      body
+    } finally {
+      SparkSession.setActiveSession(old.getOrElse(null))
+    }
+  }
```

Comment:
This in `sql` package so you should be able to access SparkSession.withActive to make it available to the `io/delta` package:

```suggestion
  def withActiveSession[T](spark: SparkSession)(body: => T): T = spark.withActive(body)
```

Also add a short comment to say this is just working around the `withActiveSession` not being visible in `io/delta`